src/main/java/lrgs/rtstat/hosts/LrgsConnection.java

```diff
@@ -0,0 +1,189 @@
+    }
+
+    public String getPassword()
+    {
+        return password;
+    }
+
+    public Date getLastUsed()
+    {
+        return lastUsed;
+    }
+
+    @Override
+    public boolean equals(Object other)
+    {
+        if (!(other instanceof LrgsConnection))
+        {
+            return false;
+        }
+        LrgsConnection rhs = (LrgsConnection)other;
+        if (port != rhs.port)
+        {
+            return false;
+        }
+
+        if (lastUsed != null && rhs.lastUsed == null)
+        {
+            return false;
+        }
+        else if (lastUsed == null && rhs.lastUsed != null)
+        {
+            return false;
+        }
+        else if ((lastUsed != null && rhs.lastUsed != null) && !lastUsed.equals(rhs.lastUsed))
+        {
+            return false;
+        }
+
+        if (!hostName.equals(rhs.hostName))
+        {
+            return false;
+        }
+
+        if (!username.equals(rhs.username))
+        {
+            return false;
+        }
+
+        if (!password.equals(rhs.password))
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(hostName, port, username, password);
+    }
+
+    @Override
+    public String toString()
+    {
+        return hostName;
+    }
+
+    public String toPropertyEntry()
+    {
+        return String.format("%d %s %d %s",
+                                port, username,
+                                (lastUsed == null ? 0 : lastUsed.getTime()),
+                                password);
+    }
+
+
+    public static LrgsConnection fromDdsFile(String host, String input)
+    {
+        final String parts[] = input.split("\\s+");
+        final int port = Integer.parseInt(parts[0]);
+        final String username = parts[1];
+        final Long lastUsed = Long.parseLong(parts[2]);
+        final String password = parts[3];
+
+        return new LrgsConnection(host, port,
+                                  username, password, new Date(lastUsed));
+    }
+
+    public static String decryptPassword(LrgsConnection c, String key)
+    {
+        final String encrypted = c.getPassword();
+        if (encrypted == null || encrypted.isEmpty())
+        {
+            return "";
+        }
+        try
+        {
+            final DesEncrypter de = new DesEncrypter(key);
+            final String password = de.decrypt(encrypted);
+            return password;
+        }
+        catch (AuthException ex)
+        {
+            log.atTrace()
+               .setCause(ex)
+               .log("Unable to decrypt password from LddsConnection files.");
+            return "";
+        }
+    }
+
+    public static String encryptPassword(String password, String key)
+    {
+        if (password == null || password.isEmpty())
+        {
+            return "";
+        }
+        try
+        {
+            final DesEncrypter de = new DesEncrypter(key);
```

Comment:
Could this custom DesEncrypter be swapped out for an Apache or similar alternative library?   Suggest research that in another PR/issue.

Reply:
Honestly, I'd rather just ditch it entirely, but it is just a wrapper around built-in tools, not a custom DES implementation so probably not worth the trouble.